### PR TITLE
Geirst/memory index support different posting list types

### PR DESCRIPTION
**Pull request by geirst**
@toregge please review
@baldersheim FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/memoryindex/posting_iterator.cpp [hunk 1]
```diff
@@ -58,80 +85,108 @@ PostingIterator<interleaved_features>::doSeek(uint32_t docId)
...
     assert(docId == _itr.getKey());
-    datastore::EntryRef featureRef(_itr.getData().get_features());
-    _featureStore.setupForUnpackFeatures(featureRef, _featureDecoder);
-    _featureDecoder.unpackFeatures(_matchData, docId);
-    setUnpacked();
+    if (unpack_normal_features) {
+        datastore::EntryRef featureRef(_itr.getData().get_features());
+        _feature_store.setupForUnpackFeatures(featureRef, _feature_decoder);
+        _feature_decoder.unpackFeatures(_matchData, docId);
+        setUnpacked();
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/memoryindex/posting_iterator.cpp]: The call to setUnpacked() should not depend on unpack_normal_features being true.

Comment (geirst) [reply searchlib/src/vespa/searchlib/memoryindex/posting_iterator.cpp]: Fixed, PTAL

File: searchlib/src/vespa/searchlib/queryeval/iterators.h [hunk 2]
```diff
@@ -16,11 +16,12 @@ class RankedSearchIteratorBase : public SearchIterator
 private:
     uint32_t _needUnpack;
 protected:
-    bool getUnpacked()       const { return _needUnpack == 0; }
     void setUnpacked()             { _needUnpack = 0; }
     void clearUnpacked()           { _needUnpack = 1; }
     uint32_t getNeedUnpack() const { return _needUnpack; }
     void incNeedUnpack()           { ++_needUnpack; }
+public:
+    bool getUnpacked()       const { return _needUnpack == 0; }
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/queryeval/iterators.h]: Remove public: below

Comment (geirst) [reply searchlib/src/vespa/searchlib/queryeval/iterators.h]: Fixed

**Review by toregge — APPROVED** [review]

**Pull request merged**